userservice/src/main/java/com/infodation/userservice/services/UserServiceImpl.java

```diff
@@ -32,11 +38,42 @@
 public class UserServiceImpl implements IUserService {
     private static final Logger logger = LoggerFactory.getLogger(UserServiceImpl.class);
     private final UserRepository userRepository;
-    public UserServiceImpl(UserRepository userRepository) {
+    private final RestTemplate restTemplate;
+
+    public UserServiceImpl(UserRepository userRepository, RestTemplate restTemplate) {
         this.userRepository = userRepository;
+        this.restTemplate = restTemplate;
+    }
+
+    @Override
+    public TaskUserResponseDTO getUserWithTasks(String userId) {
```

Comment:
Please also add try catch exception for this method because we may also encounter errors in network connection to task service